src/io/outputs/include/antares/io/outputs/columns.h

```diff
@@ -0,0 +1,188 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#pragma once
+#include <functional>
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+class IColumn
+{
+public:
+    virtual ~IColumn() = default;
+    virtual void addFromString(const std::string& value) = 0;
+    virtual std::string toString(size_t index) const = 0;
+    virtual size_t size() const = 0;
+    virtual void reserve(size_t capacity) = 0;
+    virtual void clear() = 0;
+};
+
+template<typename T>
+class TypedColumn: public IColumn
+{
+public:
+    explicit TypedColumn(std::function<std::string(const T&)> toStringFn = defaultToString,
+                         std::function<T(const std::string&)> fromStringFn = defaultFromString):
+        toStringFn_(toStringFn),
+        fromStringFn_(fromStringFn)
+    {
+    }
```

Comment:
Instead of providing `std::function`s at construction time, you can define non-pure `virtual` functions to allow derived classes to override these methods while still providing a default method.

```cpp
template<typename T>
class TypedColumn: public IColumn
{
virtual std::string toStringFn() const
{
    // paste defaultToString's implementation here
}
virtual T fromStringFn()
{
    // paste defaultFromString's implementation here
}
};
```